includes/amp-helper-functions.php

```diff
@@ -23,6 +23,11 @@ function post_supports_amp( $post ) {
 		return false;
 	}
 
+	// Listen to post meta.
+	if ( ! isset( $post->ID ) || 'disabled' === get_post_meta( $post->ID, AMP_Post_Meta_Box::POST_META_KEY, true ) ) {
+		return false;
```

Comment:
Question: instead of hard-coding a `return false` here, what if we instead here did:

```php
$skip = true;
```

If the default value for `$skip` were `false`, then this `$skip` could then be passed in below as value for the `amp_skip_post` filter. This would allow plugins to still be able to override the value.

_But_ maybe that would be a premature optimization. It would also be bad if a user indicates that they want to disable AMP for a given post, but then it still ends up getting AMP enabled on the frontend due to a filter that they don't know about.

Reply:
> But maybe that would be a premature optimization. It would also be bad if a user indicates that they want to disable AMP for a given post, but then it still ends up getting AMP enabled on the frontend due to a filter that they don't know about.

That would justified if we want to allow plugins to overwrite that admin value. That said doing that wouldn't make the admin option reflect the correct value so it would be better for plugins to use the `get_post_metadata` filter which would cover both, admin and frontend.